compiler/cpp/src/thrift/generate/t_py_generator.cc

```diff
@@ -3009,22 +3001,19 @@ THRIFT_REGISTER_GENERATOR(
     "    zope.interface:  Generate code for use with zope.interface.\n"
     "    twisted:         Generate Twisted-friendly RPC services.\n"
     "    tornado:         Generate code for use with Tornado.\n"
-    "    no_utf8strings:  Do not Encode/decode strings using utf8 in the generated code. Basically "
-    "no effect for Python 3.\n"
+    "    no_utf8strings:  Do not Encode/decode strings using utf8 in the generated code. Basically no effect for Python 3.\n"
     "    coding=CODING:   Add file encoding declare in generated file.\n"
     "    slots:           Generate code using slots for instance members.\n"
     "    dynamic:         Generate dynamic code, less code generated but slower.\n"
     "    dynbase=CLS      Derive generated classes from class CLS instead of TBase.\n"
-    "    dynfrozen=CLS    Derive generated immutable classes from class CLS instead of "
-    "TFrozenBase.\n"
+    "    dynfrozen=CLS    Derive generated immutable classes from class CLS instead of TFrozenBase.\n"
     "    dynexc=CLS       Derive generated exceptions from CLS instead of TExceptionBase.\n"
-    "    dynfrozenexc=CLS Derive generated immutable exceptions from CLS instead of "
-    "TFrozenExceptionBase.\n"
+    "    dynfrozenexc=CLS Derive generated immutable exceptions from CLS instead of TFrozenExceptionBase.\n"
     "    dynimport='from foo.bar import CLS'\n"
     "                     Add an import line to generated code to find the dynbase class.\n"
     "    package_prefix='top.package.'\n"
     "                     Package prefix for generated files.\n"
     "    old_style:       Deprecated. Generate old-style classes.\n"
-    "    enum:            Generates Python's IntEnum, connects thrift to python enums. Python 3.4 "
-    "and higher.\n"
-    "    type_hints:      Generate type hints in write method, including IntEnum generation.\n")
+    "    enum:            Generates Python's IntEnum, connects thrift to python enums. Python 3.4 and higher.\n"
+    "    type_hints:      Generate type hints and type checks in write method, including IntEnum generation.\n"
```

Comment:
hi! Given that you have updated the option to no longer default this to true. I think the doc is now lying :P 

```suggestion
    "    type_hints:      Generate type hints and type checks in write method. Requires the enum option.\n"
```

Reply:
Requires ... what happens if I don't do this?

Reply:
Youll get an exception when you try.

https://github.com/apache/thrift/pull/2929/files#diff-ad110290fb2055999f700940287b9f8db58007832084247e6ffb0bd2b14b897dR131

Reply:
Would it be better then if we change it to "implies" and do exactly that?

Reply:
That is similar to what was originally in the PR. I asked for it to be changed since the enum option can cause compatibility issues with people who do not use it historically.

Requiring you to opt into enum generation feels less prone to user error than implying it for them, in my opinion.

ofc i dont maintain this! I defer!

Reply:
ok thanks for clarifying

Reply:
Good catch thanks @salomon-smekecohen